tests/test_python_api.py

```diff
@@ -416,41 +493,41 @@ def test_python_api2():
     try:
         assert fed.core.query("broker", "something") == "#invalid"
     except AssertionError:
-        assert fed.core.query("broker", "something") == {"error": {"code": 400, "message": "unrecognized broker query"}}
+        assert fed.core.query("broker", "something") == {
+            "error": {"code": 400, "message": "unrecognized broker query"}
+        }
 
     fed.add_dependency("hello")
 
     fed.core.disconnect()
 
     assert fed.core.wait_for_disconnect()
 
-    del fed
-
     broker.disconnect()
     assert broker.wait_for_disconnect()
 
-    del broker
-
-    h.helicsCleanupLibrary()
-    h.helicsCloseLibrary()
-
 
 def test_python_api3():
     core1 = h.helicsCreateCore("inproc", "core3", "--autobroker")
 
-    assert """HelicsCore(identifier = "core3", address = "core3")""" in repr(core1)
+    assert_attributes(core1, {"identifier": "core3", "address": "core3"})
 
     core2 = core1.clone()
 
     assert core1.identifier == "core3"
 
-    source_filter1 = core1.register_filter(h.HELICS_FILTER_TYPE_DELAY, "core3SourceFilter")
+    source_filter1 = core1.register_filter(
+        h.HELICS_FILTER_TYPE_DELAY, "core3SourceFilter"
+    )
 
     source_filter1.add_source_target("ep1")
 
-    assert (
-        """<{ 'CONNECTION_REQUIRED' = 0, 'CONNECTION_OPTIONAL' = 0, 'SINGLE_CONNECTION_ONLY' = 0, 'MULTIPLE_CONNECTIONS_ALLOWED' = 0, 'BUFFER_DATA' = 0, 'RECONNECTABLE' = 0, 'STRICT_TYPE_CHECKING' = 0, 'RECEIVE_ONLY' = 0, 'SOURCE_ONLY' = 0, 'IGNORE_UNIT_MISMATCH' = 0, 'ONLY_TRANSMIT_ON_CHANGE' = 0, 'ONLY_UPDATE_ON_CHANGE' = 0, 'IGNORE_INTERRUPTS' = 0, 'MULTI_INPUT_HANDLING_METHOD' = 0, 'INPUT_PRIORITY_LOCATION' = 0, 'CLEAR_PRIORITY_LIST' = 0, 'CONNECTIONS' = 0 }>"""
-        in repr(source_filter1.option)
+    assert_entries(
+        source_filter1.option,
+        {
+            **DEFAULT_PUB_OPTION_SETTINGS,
+            h.HelicsHandleOption.MULTIPLE_CONNECTIONS_ALLOWED: 0,
```

Comment:
This makes it a lot clearer what value has changed that the assert actually cares about.